app/assets/javascripts/initializers/initializeTimeFixer.js

```diff
@@ -35,13 +33,18 @@ function convertCalEvent(utc) {
     minute: 'numeric',
   };
 
-  return getFormat(options, date);
+  return formatDateTime(options, date);
 }
 
 function updateLocal(elements, convertCallback) {
   var local;
   for (var i = 0; i < elements.length; i += 1) {
-    local = convertCallback(elements[i].dataset.datetime);
+    const utc =
+      convertCallback.name !== 'convertCalEvent'
+        ? elements[i].dataset.datetime
+        : elements[i].innerHTML;
```

Comment:
I think this makes the code less readable than before, it's also weird responsibility wise, because `updateLocal` (which should probably be called something more explicit like `updateDateTime` or something), shouldn't have any knowledge of the callback. 

I understand the need to have a single function that you can recycle but this function shouldn't have knowledge of the calling ones :D 

I think your best bet here is to have this function take a second function, which could be the "getter", which takes the element and for the first two cases extracts the `dataset.datetime` attribute and for the other one the `innerHTML`.

Something like: 

```javascript
updateLocal(utcTime, convertUtcTime, (element) => element.dataset.datetime);
updateLocal(utc, convertCalEvent, (element) => element.innerHTML);
```

I haven't tested this. Please do.

You can test all of this in the /events page, but before you need to create a published event in http://localhost:3000/internal/events

Reply:
This approach looks much better, thanks @rhymes